src/pathfinding_service/claim_fees.py

```diff
@@ -0,0 +1,132 @@
+import sys
+from typing import Iterable, Tuple
+
+import click
+import structlog
+from web3 import Web3
+from web3.contract import Contract
+
+from pathfinding_service.database import PFSDatabase
+from pathfinding_service.model import IOU
+from raiden.utils.typing import BlockNumber
+from raiden_contracts.constants import CONTRACT_ONE_TO_N
+from raiden_contracts.contract_manager import ContractManager, contracts_precompiled_path
+from raiden_libs.cli import blockchain_options, common_options, connect_to_blockchain
+from raiden_libs.types import Address
+from raiden_libs.utils import private_key_to_address
+
+log = structlog.get_logger(__name__)
+
+
+@click.command()
+@click.option(
+    '--rdn-per-eth',
+    default=0.0025,
+    type=float,
+    help='Current RDN/ETH price, used to check claimed amount > transaction cost',
+)
+@click.option(
+    '--expires-within',
+    default=4 * 60 * 24 * 7,  # one week
+    type=click.IntRange(min=1),
+    help='Only IOUs which expire withing this number of blocks will be claimed',
+)
+@common_options('raiden-monitoring-service')
+@blockchain_options
+def main(
+    private_key: str,
+    state_db: str,
+    eth_rpc: str,
+    registry_address: Address,
+    user_deposit_contract_address: Address,
+    start_block: BlockNumber,
+    confirmations: int,
+    rdn_per_eth: float,
+    expires_within: int,
+) -> None:
+    pfs_address = private_key_to_address(private_key)
+    database = PFSDatabase(filename=state_db, pfs_address=pfs_address)
+
+    web3, contract_infos = connect_to_blockchain(
+        eth_rpc=eth_rpc,
+        registry_address=registry_address,
+        user_deposit_contract_address=user_deposit_contract_address,
+        start_block=start_block,
+        # necessary so that the overwrite logic works properly
+        monitor_contract_address=Address('0x' + '1' * 40),
+        contracts_version=None,
+    )
+    contract_manager = ContractManager(contracts_precompiled_path())
+    one_to_n_contract = web3.eth.contract(
+        abi=contract_manager.get_contract_abi(CONTRACT_ONE_TO_N),
+        address=user_deposit_contract_address,
+    )
+
+    claim_cost_eth = 90897
+    claim_cost_rdn = int(claim_cost_eth / rdn_per_eth)
+    ious = list(
+        get_claimable_ious(
+            database,
+            expires_before=web3.eth.blockNumber + expires_within,
+            claim_cost_rdn=claim_cost_rdn,
+        )
+    )
+    print(f'Found {len(ious)} claimable IOUs')
+    _, failures = claim_ious(ious, claim_cost_rdn, one_to_n_contract, web3, database)
+    if failures:
+        sys.exit(1)
+
+
+def get_claimable_ious(
+    database: PFSDatabase, expires_before: int, claim_cost_rdn: int
```

Comment:
I'd type `expires_before` as a `BlockNumber` and  `claim_cost_rdn`. as a `TokenAmount`.